src/Core/src/Request.php

```diff
@@ -148,27 +154,37 @@ public function getQuery(): array
         return $this->query;
     }
 
+    public function getHostPrefix(): string
+    {
+        return $this->hostPrefix;
+    }
+
+    public function setHostPrefix(string $hostPrefix): void
+    {
+        $this->hostPrefix = $hostPrefix;
+        $this->endpoint = '';
+    }
+
     public function getEndpoint(): string
     {
         if (empty($this->endpoint)) {
             if (null === $this->parsed) {
                 throw new LogicException('Request::$endpoint must be set before using it.');
             }
 
-            $this->endpoint = $this->parsed['scheme'] . '://' . $this->parsed['host'] . (isset($this->parsed['port']) ? ':' . $this->parsed['port'] : '') . $this->uri . ($this->query ? (false === strpos($this->uri, '?') ? '?' : '&') . http_build_query($this->query, '', '&', \PHP_QUERY_RFC3986) : '');
+            $this->endpoint = $this->parsed['scheme'] . '://' . $this->hostPrefix . $this->parsed['host'] . (isset($this->parsed['port']) ? ':' . $this->parsed['port'] : '') . $this->uri . ($this->query ? (false === strpos($this->uri, '?') ? '?' : '&') . http_build_query($this->query, '', '&', \PHP_QUERY_RFC3986) : '');
         }
 
         return $this->endpoint;
     }
 
     public function setEndpoint(string $endpoint): void
     {
-        if (!empty($this->endpoint)) {
+        if (null !== $this->parsed) {
```

Comment:
this looks wrong, because endpoint is initialized to empty string.

Reply:
This is checking the `parsed` property, which is assigned in this method, not the `endpoint` one.

I removed the assignment of `$endpoint` to `$this->endpoint` in the current method, because it removed the usage of the logic inside `getEndpoint` where I injected the host prefix before the parsed host, thus ignoring the host prefix.